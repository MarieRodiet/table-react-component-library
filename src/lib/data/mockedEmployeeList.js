export const mockedList = [
  {
    FirstName: 'Julie',
    LastName: 'Champ',
    BirthDate: '12/29/1989',
    StartDate: '01/09/1992',
    Street: '7 Route de Paris',
    City: 'Eugene',
    State: 'OR',
    Zipcode: '0123',
    Department: 'Marketing',
  },
  {
    FirstName: 'Claire',
    LastName: 'Sino',
    BirthDate: '07/08/1959',
    StartDate: '01/19/2020',
    Street: 'A Great Road',
    City: 'El Paso',
    State: 'TX',
    Zipcode: '9876',
    Department: 'Sales',
  },
  {
    FirstName: 'Simon',
    LastName: 'Camus',
    BirthDate: '07/18/1909',
    StartDate: '02/09/2023',
    Street: 'Rue de la paix',
    City: 'Orlando',
    State: 'AZ',
    Zipcode: '0578',
    Department: 'Consulting',
  },
  {
    FirstName: 'Katie',
    LastName: 'Clark',
    BirthDate: '07/08/1969',
    StartDate: '02/21/2022',
    Street: 'Rue du Paradis',
    City: 'Seattle',
    State: 'WA',
    Zipcode: '0000',
    Department: 'Consulting',
  },
  {
    FirstName: 'Jacques',
    LastName: 'Thomas',
    BirthDate: '02/13/1981',
    StartDate: '07/10/1998',
    Street: '10 Grande Rue',
    City: 'Roseburg',
    State: 'CA',
    Zipcode: '07771',
    Department: 'Sales',
  },
  {
    FirstName: 'Mumu',
    LastName: 'Weim',
    BirthDate: '02/11/1999',
    StartDate: '02/31/2019',
    Street: 'Avenue Magenta',
    City: 'Springfield',
    State: 'MN',
    Zipcode: '92045',
    Department: 'Engineering',
  },
  {
    FirstName: 'Kathlyn',
    LastName: 'Willians',
    BirthDate: '10/03/1982',
    StartDate: '05/29/1997',
    Street: '254 Main Lane',
    City: 'Redmon',
    State: 'FL',
    Zipcode: '09324',
    Department: 'Marketing',
  },
  {
    FirstName: 'Shannon',
    LastName: 'Boogan',
    BirthDate: '12/23/1962',
    StartDate: '11/25/2001',
    Street: 'Pretty Court',
    City: 'Ben',
    State: 'ID',
    Zipcode: '12048',
    Department: 'Sales',
  },
  {
    FirstName: 'Kaisy',
    LastName: 'Jon',
    BirthDate: '10/08/1948',
    StartDate: '08/10/1990',
    Street: 'Fillmore Street',
    City: 'Sisters',
    State: 'OR',
    Zipcode: '97402',
    Department: 'Sales',
  },
  {
    FirstName: 'Moose',
    LastName: 'Forest',
    BirthDate: '12/16/1983',
    StartDate: '06/01/2014',
    Street: 'Happy Street',
    City: 'Plendelton',
    State: 'NY',
    Zipcode: '345234',
    Department: 'Consulting',
  },
  {
    FirstName: 'Jessica',
    LastName: 'MiddleBridge',
    BirthDate: '01/11/1978',
    StartDate: '03/17/2000',
    Street: 'Salmon Lane',
    City: 'Burley',
    State: 'KS',
    Zipcode: '2452345',
    Department: 'Sales',
  },
  {
    FirstName: 'Skippy Jon',
    LastName: 'Jones',
    BirthDate: '10/14/1988',
    StartDate: '08/21/2009',
    Street: 'Avenue Somewhere',
    City: 'Skinner',
    State: 'OH',
    Zipcode: '24624523',
    Department: 'Engineering',
  },
]

export const mockedColumns = [
  'FirstName',
  'LastName',
  'BirthDate',
  'StartDate',
  'Street',
  'City',
  'State',
  'Zipcode',
  'Department',
]
