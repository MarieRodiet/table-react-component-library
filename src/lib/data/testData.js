export const testList = [
  {
    FirstName: 'ATEST',
    LastName: 'ZTEST',
    BirthDate: '12/29/0022',
    StartDate: '01/09/1992',
    Street: 'AB',
    City: 'IJ',
    State: 'OP',
    Zipcode: '0000',
    Department: 'AAAA',
  },
  {
    FirstName: 'BTEST',
    LastName: 'YTEST',
    BirthDate: '07/08/0032',
    StartDate: '01/19/2020',
    Street: 'CD',
    City: 'KL',
    State: 'QR',
    Zipcode: '1111',
    Department: 'BBBB',
  },
  {
    FirstName: 'CTEST',
    LastName: 'WTEST',
    BirthDate: '07/18/0050',
    StartDate: '02/09/2023',
    Street: 'EF',
    City: 'MN',
    State: 'ST',
    Zipcode: '2222',
    Department: 'CCCC',
  },
]

export const testColumns = [
  {
    field: 'FirstName',
    header: 'Firstname',
    type: 'string',
    sortable: true,
    width: '80',
  },
  {
    field: 'LastName',
    header: 'Lastname',
    type: 'string',
    sortable: true,
    width: '90',
  },
  {
    field: 'BirthDate',
    header: 'Birthdate',
    type: 'date',
    sortable: true,
    width: '90',
  },
  {
    field: 'StartDate',
    header: 'Start Date',
    type: 'date',
    sortable: true,
    width: '90',
  },
  {
    field: 'Street',
    header: 'Street',
    type: 'string',
    sortable: true,
    width: '200',
  },
  {
    field: 'City',
    header: 'City',
    type: 'string',
    sortable: true,
    width: '100',
  },
  {
    field: 'State',
    header: 'State',
    type: 'string',
    sortable: false,
    width: '40',
  },
  {
    field: 'Zipcode',
    header: 'Zipcode',
    type: 'string',
    sortable: true,
    width: '70',
  },
  {
    field: 'Department',
    header: 'Department',
    type: 'string',
    sortable: true,
    width: '100',
  },
]
