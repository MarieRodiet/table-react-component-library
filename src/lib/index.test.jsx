import React from 'react'
import { describe, expect, test, afterEach } from 'vitest'
import List from './index.jsx'
import renderer from 'react-test-renderer'
import { render, screen, cleanup, fireEvent } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { testList, testColumns } from './data/testData'

describe('List', () => {
  afterEach(cleanup)

  test('Default props (Title, data and theme) work', () => {
    const { container } = render(<List />)
    expect(screen.getByText('TITLE'))

    expect(screen.getByText('Champ'))
    expect(screen.getByText('Julie'))
    expect(screen.getByText('7 Route de Paris'))
    expect(screen.getByText('Eugene'))

    const tableRows = container.querySelectorAll('.table-row-light')
    expect(tableRows.length).toBe(10)
    expect(document.querySelectorAll('.light').length).toBe(5)
  })

  test('Using debug', () => {
    render(<List data={testList} columns={testColumns} />)
    screen.debug()
  })

  test('List custom title and dark theme work', () => {
    const { container } = render(<List title="TEST TITLE" theme="dark" />)
    const tableRows = container.querySelectorAll('.table-row-dark')
    expect(tableRows.length).toBe(10)
    expect(screen.getByText('TEST TITLE'))
  })

  test('List columns and data props should render with pagination showing only the first 10 rows. Some pagination arrows should help navigate to the next and previous page', () => {
    const { container, getAllByTestId } = render(<List data={testList} columns={testColumns} />)
    const tableRows = container.querySelectorAll('.table-row-light')
    let arrowBtns = getAllByTestId('pagination-arrow-btn')
    expect(tableRows.length).toBe(10)
    expect(arrowBtns.length).toBe(2)
    //HEADERS
    expect(screen.getByText('Prenom'))
    expect(screen.getByText('Nom'))
    expect(screen.getByText('Date de Naissance'))
    expect(screen.getByText('Date de Debut'))
    expect(screen.getByText('Rue'))
    expect(screen.getByText('Ville'))
    expect(screen.getByText('Etat'))
    expect(screen.getByText('Code Postal'))
    expect(screen.getByText('Departement'))

    //FIRST DATA OBJECT
    expect(screen.getByText('ATEST'))
    expect(screen.getByText('XTEST'))
    expect(screen.getByText('12/29/0022'))
    expect(screen.getByText('01/09/1992'))
    expect(screen.getByText('AB Street'))
    expect(screen.getByText('AB City'))
    expect(screen.getByText('AL'))
    expect(screen.getByText('0000'))
    expect(screen.getByText('LLLL'))
  })

  test('It would match snapshot', () => {
    const component = renderer.create(<List data={testList} columns={testColumns} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Inserting the value of testList.FirstName in the search field should render the list with just one row', async () => {
    const { container } = render(<List data={testList} columns={testColumns} />)
    const searchInput = container.querySelector('#search-input')
    const searchButton = container.querySelector('#search-button')
    fireEvent.change(searchInput, { target: { value: 'ATEST' } })
    expect(searchInput.value).toBe('ATEST')
    await userEvent.click(searchButton)
    const tableRows = container.querySelectorAll('.table-row-light')
    expect(tableRows.length).toBe(1)
  })

  test('Clicking on a row should change its color and return the selected object', async () => {
    let s = []
    function showSelection(selection) {
      s = selection
    }
    const { container } = render(
      <List data={testList} columns={testColumns} getSelection={showSelection} />
    )
    const tableRows = container.querySelectorAll('.table-row-light')
    await userEvent.click(tableRows[0])
    const coloredCells = container.querySelectorAll('.selectedStyle')
    expect(coloredCells.length).toBe(9)
    expect(s[0].FirstName).toBe(testList[0].FirstName)
  })

  test('Selecting 20 in the Pagination should remove the pagination arrows and render all the data at once', async () => {
    const { container, getByTestId, getAllByTestId } = render(
      <List data={testList} columns={testColumns} />
    )
    let options = getAllByTestId('select-option')
    let select = getByTestId('nbOfRows-select')
    let arrowBtns = await getAllByTestId('pagination-arrow-btn')
    fireEvent.change(select, { target: { value: 20 } })
    const tableRows = container.querySelectorAll('.table-row-light')
    expect(options[0].selected).toBeFalsy()
    expect(options[1].selected).toBeTruthy()
    expect(tableRows.length).toBe(testList.length)
    expect(arrowBtns).not.toBeNull()
  })

  test('Clicking on a string header arrow should sort the columns string values', async () => {
    const { getAllByTestId } = render(<List data={testList} columns={testColumns} />)
    const sortArrows = await getAllByTestId('sort-arrow')
    fireEvent.click(sortArrows[0])
    let firstRow = getAllByTestId('cell')
    expect(firstRow[0].textContent).toBe('LTEST')
  })

  test('Clicking on a date header arrow should sort the columns date values', async () => {
    const { getAllByTestId } = render(<List data={testList} columns={testColumns} />)
    const sortArrows = await getAllByTestId('sort-arrow')
    fireEvent.click(sortArrows[3])
    let firstRow = getAllByTestId('cell')
    expect(firstRow[3].textContent).toBe('02/09/2023')
  })

  test('Clicking on a number header arrow should sort the columns number values', async () => {
    const { getAllByTestId } = render(<List data={testList} columns={testColumns} />)
    const sortArrows = await getAllByTestId('sort-arrow')
    fireEvent.click(sortArrows[7])
    let firstRow = getAllByTestId('cell')
    expect(firstRow[7].textContent).toBe('9999')
  })
})
