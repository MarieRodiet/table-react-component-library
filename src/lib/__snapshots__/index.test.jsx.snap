// Vitest Snapshot v1

exports[`List > It would match snapshot 1`] = `
<div
  className="listContainer light"
>
  <div
    className="listContainer"
  >
    <h1
      data-testid="title"
    >
      TITLE
    </h1>
    <form
      className="search"
      onSubmit={[Function]}
    >
      <label
        htmlFor="search-input"
      >
        Search
      </label>
      <input
        className="light"
        id="search-input"
        name="search"
      />
      <button
        className="light"
        id="search-button"
      >
        Search
      </button>
    </form>
    <div
      className="table-container-pagination"
    >
      <form
        className="table-container-pagination-select"
      >
        <label
          className="light"
          htmlFor="nbOfRows"
        >
          Rows per page :
        </label>
        <select
          data-testid="nbOfRows-select"
          id="nbOfRows-list"
          onChange={[Function]}
        >
          <option
            data-testid="select-option"
          >
            10
          </option>
          <option
            data-testid="select-option"
          >
            20
          </option>
          <option
            data-testid="select-option"
          >
            30
          </option>
          <option
            data-testid="select-option"
          >
            50
          </option>
          <option
            data-testid="select-option"
          >
            100
          </option>
        </select>
      </form>
      <div
        className="table-container-pagination-btnContainer light"
      >
        <button
          className="table-container-pagination-btnContainer-btn"
          data-testid="pagination-arrow-btn"
          onClick={[Function]}
        >
          <svg
            className="arrow-previous"
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M214.6 41.4c-12.5-12.5-32.8-12.5-45.3 0l-160 160c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L160 141.2V448c0 17.7 14.3 32 32 32s32-14.3 32-32V141.2L329.4 246.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3l-160-160z"
            />
          </svg>
        </button>
        <div
          className="table-container-pagination-btnContainer-currentPage"
        >
          1
        </div>
        <button
          className="table-container-pagination-btnContainer-btn"
          data-testid="pagination-arrow-btn"
          onClick={[Function]}
        >
          <svg
            className="arrow-next"
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M214.6 41.4c-12.5-12.5-32.8-12.5-45.3 0l-160 160c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L160 141.2V448c0 17.7 14.3 32 32 32s32-14.3 32-32V141.2L329.4 246.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3l-160-160z"
            />
          </svg>
        </button>
      </div>
    </div>
    <table
      className="table"
      style={
        {
          "width": "860px",
        }
      }
    >
      <thead
        className="table-header"
      >
        <tr
          className="table-header-row-light"
        >
          <th
            style={
              {
                "minWidth": "80px",
              }
            }
          >
            Prenom
            <svg
              className="sort-svg-light"
              data-testid="sort-arrow"
              onClick={[Function]}
              viewBox="0 0 256 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M145.6 7.7C141 2.8 134.7 0 128 0s-13 2.8-17.6 7.7l-104 112c-6.5 7-8.2 17.2-4.4 25.9S14.5 160 24 160H80V352H24c-9.5 0-18.2 5.7-22 14.4s-2.1 18.9 4.4 25.9l104 112c4.5 4.9 10.9 7.7 17.6 7.7s13-2.8 17.6-7.7l104-112c6.5-7 8.2-17.2 4.4-25.9s-12.5-14.4-22-14.4H176V160h56c9.5 0 18.2-5.7 22-14.4s2.1-18.9-4.4-25.9l-104-112z"
              />
            </svg>
          </th>
          <th
            style={
              {
                "minWidth": "90px",
              }
            }
          >
            Nom
            <svg
              className="sort-svg-light"
              data-testid="sort-arrow"
              onClick={[Function]}
              viewBox="0 0 256 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M145.6 7.7C141 2.8 134.7 0 128 0s-13 2.8-17.6 7.7l-104 112c-6.5 7-8.2 17.2-4.4 25.9S14.5 160 24 160H80V352H24c-9.5 0-18.2 5.7-22 14.4s-2.1 18.9 4.4 25.9l104 112c4.5 4.9 10.9 7.7 17.6 7.7s13-2.8 17.6-7.7l104-112c6.5-7 8.2-17.2 4.4-25.9s-12.5-14.4-22-14.4H176V160h56c9.5 0 18.2-5.7 22-14.4s2.1-18.9-4.4-25.9l-104-112z"
              />
            </svg>
          </th>
          <th
            style={
              {
                "minWidth": "90px",
              }
            }
          >
            Date de Naissance
            <svg
              className="sort-svg-light"
              data-testid="sort-arrow"
              onClick={[Function]}
              viewBox="0 0 256 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M145.6 7.7C141 2.8 134.7 0 128 0s-13 2.8-17.6 7.7l-104 112c-6.5 7-8.2 17.2-4.4 25.9S14.5 160 24 160H80V352H24c-9.5 0-18.2 5.7-22 14.4s-2.1 18.9 4.4 25.9l104 112c4.5 4.9 10.9 7.7 17.6 7.7s13-2.8 17.6-7.7l104-112c6.5-7 8.2-17.2 4.4-25.9s-12.5-14.4-22-14.4H176V160h56c9.5 0 18.2-5.7 22-14.4s2.1-18.9-4.4-25.9l-104-112z"
              />
            </svg>
          </th>
          <th
            style={
              {
                "minWidth": "90px",
              }
            }
          >
            Date de Debut
            <svg
              className="sort-svg-light"
              data-testid="sort-arrow"
              onClick={[Function]}
              viewBox="0 0 256 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M145.6 7.7C141 2.8 134.7 0 128 0s-13 2.8-17.6 7.7l-104 112c-6.5 7-8.2 17.2-4.4 25.9S14.5 160 24 160H80V352H24c-9.5 0-18.2 5.7-22 14.4s-2.1 18.9 4.4 25.9l104 112c4.5 4.9 10.9 7.7 17.6 7.7s13-2.8 17.6-7.7l104-112c6.5-7 8.2-17.2 4.4-25.9s-12.5-14.4-22-14.4H176V160h56c9.5 0 18.2-5.7 22-14.4s2.1-18.9-4.4-25.9l-104-112z"
              />
            </svg>
          </th>
          <th
            style={
              {
                "minWidth": "200px",
              }
            }
          >
            Rue
            <svg
              className="sort-svg-light"
              data-testid="sort-arrow"
              onClick={[Function]}
              viewBox="0 0 256 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M145.6 7.7C141 2.8 134.7 0 128 0s-13 2.8-17.6 7.7l-104 112c-6.5 7-8.2 17.2-4.4 25.9S14.5 160 24 160H80V352H24c-9.5 0-18.2 5.7-22 14.4s-2.1 18.9 4.4 25.9l104 112c4.5 4.9 10.9 7.7 17.6 7.7s13-2.8 17.6-7.7l104-112c6.5-7 8.2-17.2 4.4-25.9s-12.5-14.4-22-14.4H176V160h56c9.5 0 18.2-5.7 22-14.4s2.1-18.9-4.4-25.9l-104-112z"
              />
            </svg>
          </th>
          <th
            style={
              {
                "minWidth": "100px",
              }
            }
          >
            Ville
            <svg
              className="sort-svg-light"
              data-testid="sort-arrow"
              onClick={[Function]}
              viewBox="0 0 256 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M145.6 7.7C141 2.8 134.7 0 128 0s-13 2.8-17.6 7.7l-104 112c-6.5 7-8.2 17.2-4.4 25.9S14.5 160 24 160H80V352H24c-9.5 0-18.2 5.7-22 14.4s-2.1 18.9 4.4 25.9l104 112c4.5 4.9 10.9 7.7 17.6 7.7s13-2.8 17.6-7.7l104-112c6.5-7 8.2-17.2 4.4-25.9s-12.5-14.4-22-14.4H176V160h56c9.5 0 18.2-5.7 22-14.4s2.1-18.9-4.4-25.9l-104-112z"
              />
            </svg>
          </th>
          <th
            style={
              {
                "minWidth": "40px",
              }
            }
          >
            Etat
            <svg
              className="sort-svg-light"
              data-testid="sort-arrow"
              onClick={[Function]}
              viewBox="0 0 256 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M145.6 7.7C141 2.8 134.7 0 128 0s-13 2.8-17.6 7.7l-104 112c-6.5 7-8.2 17.2-4.4 25.9S14.5 160 24 160H80V352H24c-9.5 0-18.2 5.7-22 14.4s-2.1 18.9 4.4 25.9l104 112c4.5 4.9 10.9 7.7 17.6 7.7s13-2.8 17.6-7.7l104-112c6.5-7 8.2-17.2 4.4-25.9s-12.5-14.4-22-14.4H176V160h56c9.5 0 18.2-5.7 22-14.4s2.1-18.9-4.4-25.9l-104-112z"
              />
            </svg>
          </th>
          <th
            style={
              {
                "minWidth": "70px",
              }
            }
          >
            Code Postal
            <svg
              className="sort-svg-light"
              data-testid="sort-arrow"
              onClick={[Function]}
              viewBox="0 0 256 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M145.6 7.7C141 2.8 134.7 0 128 0s-13 2.8-17.6 7.7l-104 112c-6.5 7-8.2 17.2-4.4 25.9S14.5 160 24 160H80V352H24c-9.5 0-18.2 5.7-22 14.4s-2.1 18.9 4.4 25.9l104 112c4.5 4.9 10.9 7.7 17.6 7.7s13-2.8 17.6-7.7l104-112c6.5-7 8.2-17.2 4.4-25.9s-12.5-14.4-22-14.4H176V160h56c9.5 0 18.2-5.7 22-14.4s2.1-18.9-4.4-25.9l-104-112z"
              />
            </svg>
          </th>
          <th
            style={
              {
                "minWidth": "100px",
              }
            }
          >
            Departement
            <svg
              className="sort-svg-light"
              data-testid="sort-arrow"
              onClick={[Function]}
              viewBox="0 0 256 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M145.6 7.7C141 2.8 134.7 0 128 0s-13 2.8-17.6 7.7l-104 112c-6.5 7-8.2 17.2-4.4 25.9S14.5 160 24 160H80V352H24c-9.5 0-18.2 5.7-22 14.4s-2.1 18.9 4.4 25.9l104 112c4.5 4.9 10.9 7.7 17.6 7.7s13-2.8 17.6-7.7l104-112c6.5-7 8.2-17.2 4.4-25.9s-12.5-14.4-22-14.4H176V160h56c9.5 0 18.2-5.7 22-14.4s2.1-18.9-4.4-25.9l-104-112z"
              />
            </svg>
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          className="table-row-light"
          onClick={[Function]}
        >
          <td
            className=""
            data-testid="cell"
          >
            ATEST
          </td>
          <td
            className=""
            data-testid="cell"
          >
            XTEST
          </td>
          <td
            className=""
            data-testid="cell"
          >
            12/29/0022
          </td>
          <td
            className=""
            data-testid="cell"
          >
            01/09/1992
          </td>
          <td
            className=""
            data-testid="cell"
          >
            AB Street
          </td>
          <td
            className=""
            data-testid="cell"
          >
            AB City
          </td>
          <td
            className=""
            data-testid="cell"
          >
            AL
          </td>
          <td
            className=""
            data-testid="cell"
          >
            0000
          </td>
          <td
            className=""
            data-testid="cell"
          >
            LLLL
          </td>
        </tr>
        <tr
          className="table-row-light"
          onClick={[Function]}
        >
          <td
            className=""
            data-testid="cell"
          >
            BTEST
          </td>
          <td
            className=""
            data-testid="cell"
          >
            WTEST
          </td>
          <td
            className=""
            data-testid="cell"
          >
            07/08/0032
          </td>
          <td
            className=""
            data-testid="cell"
          >
            01/19/2020
          </td>
          <td
            className=""
            data-testid="cell"
          >
            CD Street
          </td>
          <td
            className=""
            data-testid="cell"
          >
            CD  City
          </td>
          <td
            className=""
            data-testid="cell"
          >
            AK
          </td>
          <td
            className=""
            data-testid="cell"
          >
            1111
          </td>
          <td
            className=""
            data-testid="cell"
          >
            KKKK
          </td>
        </tr>
        <tr
          className="table-row-light"
          onClick={[Function]}
        >
          <td
            className=""
            data-testid="cell"
          >
            CTEST
          </td>
          <td
            className=""
            data-testid="cell"
          >
            VTEST
          </td>
          <td
            className=""
            data-testid="cell"
          >
            07/18/0050
          </td>
          <td
            className=""
            data-testid="cell"
          >
            02/09/2023
          </td>
          <td
            className=""
            data-testid="cell"
          >
            EF Street
          </td>
          <td
            className=""
            data-testid="cell"
          >
            EF  City
          </td>
          <td
            className=""
            data-testid="cell"
          >
            AZ
          </td>
          <td
            className=""
            data-testid="cell"
          >
            2222
          </td>
          <td
            className=""
            data-testid="cell"
          >
            JJJJ
          </td>
        </tr>
        <tr
          className="table-row-light"
          onClick={[Function]}
        >
          <td
            className=""
            data-testid="cell"
          >
            DTEST
          </td>
          <td
            className=""
            data-testid="cell"
          >
            UTEST
          </td>
          <td
            className=""
            data-testid="cell"
          >
            02/11/1999
          </td>
          <td
            className=""
            data-testid="cell"
          >
            02/31/2019
          </td>
          <td
            className=""
            data-testid="cell"
          >
            GH Street
          </td>
          <td
            className=""
            data-testid="cell"
          >
            GH City
          </td>
          <td
            className=""
            data-testid="cell"
          >
            AR
          </td>
          <td
            className=""
            data-testid="cell"
          >
            3333
          </td>
          <td
            className=""
            data-testid="cell"
          >
            IIII
          </td>
        </tr>
        <tr
          className="table-row-light"
          onClick={[Function]}
        >
          <td
            className=""
            data-testid="cell"
          >
            ETEST
          </td>
          <td
            className=""
            data-testid="cell"
          >
            TTEST
          </td>
          <td
            className=""
            data-testid="cell"
          >
            02/13/1981
          </td>
          <td
            className=""
            data-testid="cell"
          >
            07/10/1998
          </td>
          <td
            className=""
            data-testid="cell"
          >
            IJ Street
          </td>
          <td
            className=""
            data-testid="cell"
          >
            IJ City
          </td>
          <td
            className=""
            data-testid="cell"
          >
            CA
          </td>
          <td
            className=""
            data-testid="cell"
          >
            4444
          </td>
          <td
            className=""
            data-testid="cell"
          >
            HHHH
          </td>
        </tr>
        <tr
          className="table-row-light"
          onClick={[Function]}
        >
          <td
            className=""
            data-testid="cell"
          >
            FTEST
          </td>
          <td
            className=""
            data-testid="cell"
          >
            STEST
          </td>
          <td
            className=""
            data-testid="cell"
          >
            10/03/1982
          </td>
          <td
            className=""
            data-testid="cell"
          >
            05/29/1997
          </td>
          <td
            className=""
            data-testid="cell"
          >
            KL Street
          </td>
          <td
            className=""
            data-testid="cell"
          >
            KL City
          </td>
          <td
            className=""
            data-testid="cell"
          >
            CO
          </td>
          <td
            className=""
            data-testid="cell"
          >
            5555
          </td>
          <td
            className=""
            data-testid="cell"
          >
            GGGG
          </td>
        </tr>
        <tr
          className="table-row-light"
          onClick={[Function]}
        >
          <td
            className=""
            data-testid="cell"
          >
            GTEST
          </td>
          <td
            className=""
            data-testid="cell"
          >
            RTEST
          </td>
          <td
            className=""
            data-testid="cell"
          >
            12/23/1962
          </td>
          <td
            className=""
            data-testid="cell"
          >
            11/25/2001
          </td>
          <td
            className=""
            data-testid="cell"
          >
            MN Street
          </td>
          <td
            className=""
            data-testid="cell"
          >
            MN City
          </td>
          <td
            className=""
            data-testid="cell"
          >
            CT
          </td>
          <td
            className=""
            data-testid="cell"
          >
            6666
          </td>
          <td
            className=""
            data-testid="cell"
          >
            FFFF
          </td>
        </tr>
        <tr
          className="table-row-light"
          onClick={[Function]}
        >
          <td
            className=""
            data-testid="cell"
          >
            HTEST
          </td>
          <td
            className=""
            data-testid="cell"
          >
            QTEST
          </td>
          <td
            className=""
            data-testid="cell"
          >
            10/08/1948
          </td>
          <td
            className=""
            data-testid="cell"
          >
            08/10/1990
          </td>
          <td
            className=""
            data-testid="cell"
          >
            OP Street
          </td>
          <td
            className=""
            data-testid="cell"
          >
            OP City
          </td>
          <td
            className=""
            data-testid="cell"
          >
            ID
          </td>
          <td
            className=""
            data-testid="cell"
          >
            7777
          </td>
          <td
            className=""
            data-testid="cell"
          >
            EEEE
          </td>
        </tr>
        <tr
          className="table-row-light"
          onClick={[Function]}
        >
          <td
            className=""
            data-testid="cell"
          >
            ITEST
          </td>
          <td
            className=""
            data-testid="cell"
          >
            PTEST
          </td>
          <td
            className=""
            data-testid="cell"
          >
            12/16/1983
          </td>
          <td
            className=""
            data-testid="cell"
          >
            06/01/2014
          </td>
          <td
            className=""
            data-testid="cell"
          >
            QR Street
          </td>
          <td
            className=""
            data-testid="cell"
          >
            QR City
          </td>
          <td
            className=""
            data-testid="cell"
          >
            IA
          </td>
          <td
            className=""
            data-testid="cell"
          >
            8888
          </td>
          <td
            className=""
            data-testid="cell"
          >
            DDDD
          </td>
        </tr>
        <tr
          className="table-row-light"
          onClick={[Function]}
        >
          <td
            className=""
            data-testid="cell"
          >
            JTEST
          </td>
          <td
            className=""
            data-testid="cell"
          >
            OTEST
          </td>
          <td
            className=""
            data-testid="cell"
          >
            01/11/1978
          </td>
          <td
            className=""
            data-testid="cell"
          >
            03/17/2000
          </td>
          <td
            className=""
            data-testid="cell"
          >
            ST Street
          </td>
          <td
            className=""
            data-testid="cell"
          >
            ST City
          </td>
          <td
            className=""
            data-testid="cell"
          >
            KS
          </td>
          <td
            className=""
            data-testid="cell"
          >
            9999
          </td>
          <td
            className=""
            data-testid="cell"
          >
            CCCC
          </td>
        </tr>
        <tr
          className="table-row-light"
          onClick={[Function]}
        >
          <td
            className=""
            data-testid="cell"
          >
            KTEST
          </td>
          <td
            className=""
            data-testid="cell"
          >
            NTEST
          </td>
          <td
            className=""
            data-testid="cell"
          >
            10/14/1988
          </td>
          <td
            className=""
            data-testid="cell"
          >
            08/21/2009
          </td>
          <td
            className=""
            data-testid="cell"
          >
            UV Street
          </td>
          <td
            className=""
            data-testid="cell"
          >
            UV City
          </td>
          <td
            className=""
            data-testid="cell"
          >
            WA
          </td>
          <td
            className=""
            data-testid="cell"
          >
            1010
          </td>
          <td
            className=""
            data-testid="cell"
          >
            BBBB
          </td>
        </tr>
        <tr
          className="table-row-light"
          onClick={[Function]}
        >
          <td
            className=""
            data-testid="cell"
          >
            LTEST
          </td>
          <td
            className=""
            data-testid="cell"
          >
            MTEST
          </td>
          <td
            className=""
            data-testid="cell"
          >
            07/18/1050
          </td>
          <td
            className=""
            data-testid="cell"
          >
            02/09/2013
          </td>
          <td
            className=""
            data-testid="cell"
          >
            WX Street
          </td>
          <td
            className=""
            data-testid="cell"
          >
            WX City
          </td>
          <td
            className=""
            data-testid="cell"
          >
            WI
          </td>
          <td
            className=""
            data-testid="cell"
          >
            1212
          </td>
          <td
            className=""
            data-testid="cell"
          >
            AAAA
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
